source/js/searcher.js

```diff
@@ -0,0 +1,771 @@
+     * @param {Boolean} noTag Check text only; ignoring html tags. (false)
+     * @return {String} The content of the website.
+     */
+    getSiteContent: (htmlText, contentSearchQuery, noTag = false) => {
+        
+        //Create pseudo html document in memory.
+        let doc = document.createElement("html");
+        
+        doc.innerHTML = htmlText;
+
+        //Quick get round for image link error. Remove image src attribute.
+        let imgs = doc.getElementsByTagName("img");
+        for (i in imgs) {
+            let img = imgs[i];
+            img.src = "";
+        }
+        
+        //Select the content using the search query provided.
+        let contEl = doc.querySelector(contentSearchQuery);
+        if (!contEl) return "";
+        if (noTag) return contEl.innerText;
+        return contEl.innerHTML;
+    },
+    /**
+     * Method used for getting the string content of the provided string, based on the starting position and ending position.
+     * @param {String} text The provided string.
+     * @param {Number} start The zero-based index for the starting position in the text.
+     * @param {Number} end The number representing the number of characters removed from the end of the text.
+     * @returns {String} The content of the text.
+     */
+    getStringContent: (text, start, end = 0) => {
+        let startString = text.substring(start);
+        return startString.substr(0, startString.length - end);
+    },
+    /**
+     * Method used for getting the search value from the search query in the url.
+     * @param {String} stringQuery The string query flag to search for in order to get the search value.
+     * @returns {String} The searched value.
+     */
+    getSiteGetRequestData: (stringQuery) => {
+        const searchQuery = location.search;
+        if (F.isEmpty(searchQuery)) return null;
+
+        let rExp = new RegExp(`(\\?|\\&)(${stringQuery}).+?\\&{1,}.+?\\b\\=|(\\?|\\&)(${stringQuery}).+`, "gm");
+        let splitSQ = searchQuery.match(rExp);
+        if (splitSQ) {
+            let fRes = splitSQ[0];
+            rExp = /\&[\w]{1,}\=/;
+            fRes = fRes.split(rExp)[0];
+            return F.getRequestDataParser(F.getStringContent(fRes, stringQuery.length + 2, 0));
+        }
+        return "";
+    },
+    /**
+     * Computes a new string in which certain characters have been replaced by a hexadecimal escape sequence.
+     * @param {String} text Text to be escaped.
+     * @returns {String} New string with replaced characters.
+     */
+    escapeRegex: (text) => {
+        return text.replace(/[-\/\\^$*+?.()|[\]{}]/gm, "\\$1");
+    },
+    /**
+     * Method used for getting predicted name from the url provided based on the root directory of the websites.
+     * @param {String} url The url provided.
+     * @param {String} rootDir The root directory of the website.
+     * @returns {String} The predicted name of the site.
+     */
+    getPossibleSiteName: (url, rootDir) => {
```

Comment:
this function is a bit hard to read - maybe use clearer variable names and add more explanation?

Reply:
Added comments and comments that explain the names of the variables.